torch/_inductor/compile_fx.py

```diff
@@ -587,6 +587,11 @@ def _shape_env_from_inputs(inputs):
     if fake_mode is not None:
         return fake_mode.shape_env
 
+    # When there are no tensor inputs, get shape_env from the first SymInt.
+    for input in inputs:
+        if isinstance(input, torch.SymInt):
+            return input.node.shape_env
```

Comment:
Will extend this to support the other `Sym*` types that we have. Will add additional tests.

Reply:
Why this change: `test_div8` has only non-tensor args. Before, this returned `None` for `shape_env` in `compile_fx_inner`, where `shape_env` is copied to `GraphLowering`. Then, `var_to_val` in `size_hint` was empty and there was no way to associate vars with input values.